File: backend/src/main/java/se/matildaerenius/fluxo/domain/dto/TaskDto.java
package se.matildaerenius.fluxo.domain.dto;

import se.matildaerenius.fluxo.domain.entities.TaskPriority;
import se.matildaerenius.fluxo.domain.entities.TaskStatus;

import java.time.LocalDateTime;
import java.util.UUID;

public record TaskDto(
        UUID id,
        String title,
        String description,
        LocalDateTime dueDate,
        TaskPriority priority,
        TaskStatus status
) {
}

File: android-app/app/src/main/kotlin/se/matildaerenius/fluxo/data/remote/dto/TaskDto.kt
package se.matildaerenius.fluxo.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class TaskDto(
    val id: String? = null,
    val title: String,
    val isCompleted: Boolean = false,
    val priority: Int = 1
)
